script.js

```diff
@@ -1,12 +1,92 @@
 //You can edit ALL of the code here
+//Variables
+const rootElem = document.getElementById("root");
+const allEpisodes = getAllEpisodes();
+const selectEpisode = document.getElementById("allEpisodes");
+let searchBox = document.getElementById("search");
 function setup() {
-  const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
+  listEpisodes(allEpisodes);
+  // Item search function by using an event Listener
+  searchBox.addEventListener("input", () => {
+    let resultNumber = document.getElementById("resultNumber");
+    let results = allEpisodes.filter(containsSearchTerm);
+    makePageForEpisodes(results);
+    resultNumber.innerText = `${results.length} / ${allEpisodes.length}`;
+    if (searchBox.value.length == 0) {
+      resultNumber.innerText = "";
+    }
+  });
+  selectEpisode.addEventListener("change", () => {
+    if (selectEpisode.value === "novalue") {
+      makePageForEpisodes(allEpisodes);
+    } else {
+      let result = allEpisodes.find((episode) => {
+        return episode.id == selectEpisode.value;
+      });
+      makePageForEpisodes(result);
+    }
+  });
 }
-
+// Episodes display function
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  rootElem.innerHTML = ""; // Clearing the main container
+  for (let i = 0; i < episodeList.length; i++) {
+    // Variables to be used for each episode
+    let episodeContainer = document.createElement("div"),
+      episodeShadow = document.createElement("div"),
+      episodeBody = document.createElement("div"),
```

Comment:
This logic is quite bulky at the moment - you might want to consider splitting it up into a separate function that you can reuse inside the `for` loop